microsetta_interface/tests/test_integration.py

```diff
@@ -0,0 +1,161 @@
+import unittest
+import hashlib
+import os
+import uuid
+from unittest.mock import patch
+import jwt
+from microsetta_interface.server import app
+from microsetta_interface.implementation import TOKEN_KEY_NAME
+from microsetta_interface.config_manager import SERVER_CONFIG
+import requests
+
+
+# check if microsetta-private-api appears to be running
+PRIVATE_API_AVAILABLE = False
+try:
+    req = requests.get(SERVER_CONFIG['private_api_endpoint'] + '/ui')
+except:  # noqa
+    PRIVATE_API_AVAILABLE = False
+else:
+    PRIVATE_API_AVAILABLE = (req.status_code == 200)
+
+
+# obtain the JWT key if it exists
+PRIVKEY_ENVVAR = 'MICROSETTA_INTERFACE_DEBUG_JWT_PRIV'
+if os.environ.get(PRIVKEY_ENVVAR, False):
+    PRIV_KEY = open(os.environ[PRIVKEY_ENVVAR]).read()
+else:
+    PRIVATE_API_AVAILABLE = False
+
+
+def _fake_jwt(email, verified):
+    if PRIVATE_API_AVAILABLE:
+        # lie and say we're from authrocket
+        payload = {'email': email,
+                   'email_verified': verified,
+                   'iss': 'https://authrocket.com',
+                   'sub': hashlib.md5(email.encode('ascii')).hexdigest()}
+        encoded = jwt.encode(payload, PRIV_KEY, algorithm='RS256')
+        return encoded
+
+
+# we cannot have fine grain control over the state of the private-api
+# during integration testing. we also do not have control over the
+# order in which tests are executed. as such, treat each user with
+# care, and define new users on an as need basis.
+USER_NEW_EMAIL = 'user_new@foo.bar'
+USER_NEW = _fake_jwt(USER_NEW_EMAIL, True)
+
+USER_NEW_UNVERIFIED_EMAIL = 'user_unverified@foo.bar'
+USER_NEW_UNVERIFIED = _fake_jwt(USER_NEW_UNVERIFIED_EMAIL, False)
+
+USER_WITH_VALID_SAMPLE_EMAIL = "th-dq)wort@3'rey.i3l"
+USER_WITH_VALID_SAMPLE = _fake_jwt(USER_WITH_VALID_SAMPLE_EMAIL, True)
+
+
+@unittest.skipIf(not PRIVATE_API_AVAILABLE,
```

Comment:
Do you consider not having private api available as a failure?  I feel this might silently pass if we break something in the install scripts.

Reply:
No. The other unit tests and assessments of the repo can be done independent of -private-api, and it would be annoying to always run -private-api locally. 

Note that the actions script in the integration test job examines the output of this run, and errors if tests were skipped

Reply:
Just to note, if the install scripts are broken, we should observe a nonzero exit status on start of private-api. I'm going to close this out as well since we are testing job output to check for skips.